source/braille.py

```diff
@@ -2762,6 +2801,11 @@ def terminate():
 	global handler
 	handler.terminate()
 	handler = None
+	# noqa: F401 avoid module level import to prevent cyclical dependency
+	# between speech and braille
+	from speech.extensions import pre_speech, pre_speechCanceled

```

Comment:
Please move these unregistrations to the top of this function. We need to ensure that these extension points are never called with an unitialized handler. As a general rule of thumb, it makes sense to mirror calls to the initialize function. So if the initialize does initializeA, initializeB, initializeC, terminate should do terminateC, terminateB, terminateA.